pkg/server/sandbox_run.go

```diff
@@ -498,14 +487,36 @@ func (c *criContainerdService) unmountSandboxFiles(rootDir string, config *runti
 	return nil
 }
 
+// setupPod setups up the network for a pod
+func (c *criContainerdService) setupPod(id string, path string, config *runtime.PodSandboxConfig) (string, error) {
+	labels := map[string]string{
+		"K8S_POD_NAMESPACE":          config.GetMetadata().GetNamespace(),
+		"K8S_POD_NAME":               config.GetMetadata().GetName(),
+		"K8S_POD_INFRA_CONTAINER_ID": id,
+		"IgnoreUnknown":              "1",
+	}
+	result, err := c.netPlugin.Setup(id,
+		path,
+		cni.WithLabels(labels),
+		cni.WithCapabilityPortMap(toCNIPortMappings(config.GetPortMappings())))
+	if err != nil {
+		return "", err
+	}
+	// Check if the default interface has IP config
+	if configs, ok := result.Interfaces[defaultIfName]; ok && len(configs.IPConfigs) > 0 {
+		return configs.IPConfigs[0].IP.String(), nil
+	}
+	return "", fmt.Errorf("failed to find network info for sandbox %q", id)
```

Comment:
Do we need to remove on error?

Reply:
The failure here would mean the cni plugin is not allocating the IP and hence missing in the result.

Reply:
I mean, do we need a `defer teardown` on error?

Reply:
it is takes care of by the called RunPodSandbox

Reply:
No, that defer is after this error.

Reply:
ok i se ewhat u mean